{"@class": "de.fau.cs.inf2.cthree.data.Cluster", "id": "ec3be244f88d8b4c5a71870d6f8ef858", "detectedBy": ["DIFF_HIERARCHICAL"], "members": [{"@class": "de.fau.cs.inf2.cthree.data.CodeChange", "id": "46569d521ee63f8253649c8df04dc380", "repository": "http://git.eclipse.org/gitroot/jdt/eclipse.jdt.core.git", "fileName": "org.eclipse.jdt.core/compiler/org/eclipse/jdt/internal/compiler/ReadManager.java", "commitBeforeChange": "48628a48bc30e9f74feb6b693bbfe398c063e4b8", "commitAfterChange": "fea4095ba208aac8a9915b5d68b0792beec39b62", "methodNumberBeforeChange": 1, "methodNumberAfterChange": 1, "signatureBeforeChange": "  public char[] getContents(ICompilationUnit unit)", "signatureAfterChange": "  public char[] getContents(ICompilationUnit unit) throws Error", "diff": ["-public char[] getContents(ICompilationUnit unit) {", "-\tif (this.readingThreads == null || this.units.length == 0)", "+public char[] getContents(ICompilationUnit unit) throws Error {", "+\tif (this.readingThreads == null || this.units.length == 0) {", "+\t\tif (this.caughtException != null) {", "+\t\t\t// rethrow the caught exception from the readingThreads in the main compiler thread", "+\t\t\tif (caughtException instanceof Error)", "+\t\t\t\tthrow (Error) caughtException;", "+\t\t\tthrow (RuntimeException) caughtException;", "+\t\t}", "+\t}"]}, {"@class": "de.fau.cs.inf2.cthree.data.CodeChange", "id": "cf40857f7413e4e6d224de13cab152e6", "repository": "http://git.eclipse.org/gitroot/jdt/eclipse.jdt.core.git", "fileName": "org.eclipse.jdt.core/compiler/org/eclipse/jdt/internal/compiler/ProcessTaskManager.java", "commitBeforeChange": "48628a48bc30e9f74feb6b693bbfe398c063e4b8", "commitAfterChange": "fea4095ba208aac8a9915b5d68b0792beec39b62", "methodNumberBeforeChange": 2, "methodNumberAfterChange": 2, "signatureBeforeChange": "  public CompilationUnitDeclaration removeNextUnit()", "signatureAfterChange": "  public CompilationUnitDeclaration removeNextUnit() throws Error", "diff": ["-public CompilationUnitDeclaration removeNextUnit() {", "+public CompilationUnitDeclaration removeNextUnit() throws Error {", "-\t\tif (next == null) {", "+\t\tif (next == null || this.caughtException != null) {", "-\t\t\t\tif (this.processingThread == null)", "+\t\t\t\tif (this.processingThread == null) {", "+\t\t\t\t\tif (this.caughtException != null) {", "+\t\t\t\t\t\t// rethrow the caught exception from the processingThread in the main compiler thread", "+\t\t\t\t\t\tif (caughtException instanceof Error)", "+\t\t\t\t\t\t\tthrow (Error) caughtException;", "+\t\t\t\t\t\tthrow (RuntimeException) caughtException;", "+\t\t\t\t\t}", "+\t\t\t\t}"]}], "num": 47314}